art/_config.py

```diff
@@ -2,20 +2,28 @@
 
 from __future__ import absolute_import
 
-import re
 import os
 from . import _paths
 from . import _yaml
 
 
-def save(gitlab_url, private_token):
+def save(gitlab_url, token_type, token):
     config = {
         'gitlab_url': gitlab_url,
-        'private_token': private_token,
+        'token_type': token_type,
+        'token': token,
     }
     _paths.mkdirs(os.path.dirname(_paths.config_file))
     _yaml.save(_paths.config_file, config)
 
 
 def load():
-    return _yaml.load(_paths.config_file)
+    config = _yaml.load(_paths.config_file)
+   
+    # migrate legacy private_token to token/token_type
+    if 'private_token' in config:
+        config['token'] = config['private_token']
+        config['token_type'] = 'private'
```

Comment:
There's an annoying corner case here:
```
private_token = abcdefgh
token_type = job
```

I think this is actually likely in the real world when a user starts from an existing config with `private_token`. If they change the token value (but not the key) and add `token_type = job`, it will get silently overwritten, leaving the user confused.

Probably better to check if `token_type` is set and abort.

Also, maybe issue a `warning.warning` suggesting they migrate?

Reply:
I see your point, and it's correct, but no one can put a job token in their config file. They're ephemeral.

Reply:
Oh right, I totally overlooked that. Probably not worth worrying about then.

Reply:
I don't want someone to lose their token if they provide both private_token and token, so I'll improve the migration and add some validation to protect against the case where both keys are provided.

Reply:
Reworked the migration and added validation (including the specific warning you suggested) in 81d971412695ead0c2eb4178d8eeb51ef9396b08